pkg/controller/cluster/controller.go

```diff
@@ -23,33 +23,86 @@ import (
 
 	"github.com/kubernetes-incubator/apiserver-builder/pkg/builders"
 
-	"sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset/typed/cluster/v1alpha1"
 	"sigs.k8s.io/cluster-api/pkg/controller/sharedinformers"
 	listers "sigs.k8s.io/cluster-api/pkg/client/listers_generated/cluster/v1alpha1"
+	clusterv1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"github.com/golang/glog"
+	"sigs.k8s.io/cluster-api/pkg/client/clientset_generated/clientset"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/cluster-api/util"
 )
 
 // +controller:group=cluster,version=v1alpha1,kind=Cluster,resource=clusters
 type ClusterControllerImpl struct {
 	builders.DefaultControllerFns
 
+	actuator Actuator
 	// lister indexes properties about Cluster
 	lister listers.ClusterLister
+
+	kubernetesClientSet *kubernetes.Clientset
+	clientSet           clientset.Interface
+	clusterClient       v1alpha1.ClusterInterface
 }
 
 // Init initializes the controller and is called by the generated code
```

Comment:
Please add a Run command that calls cache.WaitForSync

Point 6 of the Writing Controllers guide, https://github.com/kubernetes/community/blob/master/contributors/devel/controllers.md

Related to this issue as well, https://github.com/kubernetes-sigs/cluster-api/issues/106

You can see this PR for how I try to do it in machineset controller, https://github.com/kubernetes-sigs/cluster-api/pull/186/files#diff-b0f93c81bd86491a95181a762269c619R93

Reply:
I was planning to move the code that deploys things like a default storage class, secrets for service account keys and the machine controller deployment into the cluster controller. Do you think those are secondary resources that I should wait to sync too?

Reply:
I think the idea is to sync on the things that you have listers for. It would be fine to add additional synced InformerSynced once you add the logic for them.
The machine lister synced is not necessary until you start using the machine lister.

The idea with WaitForSync is to ensure that the listers which are backed by caches are up to date.

Reply:
removed machine lister for now, we might need it later but you're right that it isn't necessary in this PR